dmoj/cptbox/filesystem_policies.py

```diff
@@ -0,0 +1,113 @@
+import os
+from enum import Enum
+from typing import List, Union
+
+
+class AccessMode(Enum):
+    NONE = 0
+    EXACT = 1
+    RECURSIVE = 2
+
+
+class Dir:
+    def __init__(self):
+        self.access_mode = AccessMode.NONE
+        self.subdirectory_map = {}
```

Comment:
Mmm, is this actually a map of subdirectories? My reading of the code suggests it can contain `File`s too.